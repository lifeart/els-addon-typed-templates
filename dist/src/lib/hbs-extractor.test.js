"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const ast_parser_1 = require("./ast-parser");
const hbs_extractor_1 = require("./hbs-extractor");
function results(source) {
    let items = hbs_extractor_1.extractRelationships(ast_parser_1.getClassMeta(source).nodes);
    delete items.klass;
    return items;
}
describe('getClassMeta', () => {
    const cases = [
        '{{1}}',
        `{{"1"}}`,
        `{{foo}}`,
        `{{#let foo as |bar|}}{{bar}}{{/let}}`,
        '<MyComponent as |ctx|> {{ctx.result}} </MyComponent>',
        `{{foo this.bar baz=this.boo}}`,
        '<input {{foo-bar this.baz a="1"}}>',
        '<MyComponent as |ctx|>{{#let ctx.result as |bar|}}{{bar}}{{/let}}</MyComponent>',
    ];
    cases.forEach((str) => {
        it(str, () => {
            expect(results(str)).toMatchSnapshot();
        });
    });
});
//# sourceMappingURL=hbs-extractor.test.js.map