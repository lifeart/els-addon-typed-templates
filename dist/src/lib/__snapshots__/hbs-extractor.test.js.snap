// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "blockPaths": Array [
    "2,7:2,11 - PathExpression",
    "3,2:7,11 - PathExpression",
  ],
  "componentsForImport": Array [
    "link-to",
  ],
  "parents": Object {
    "2,4:8,9 - BlockStatement": Array [
      "3,2:7,11 - BlockStatement",
      "6,4:6,11 - MustacheStatement",
      "2,7:2,11 - PathExpression",
      "2,12:2,27 - PathExpression",
    ],
    "3,2:7,11 - BlockStatement": Array [
      "6,4:6,11 - MustacheStatement",
      "3,2:7,11 - PathExpression",
      "4,11:4,28 - SubExpression",
      "4,23:4,26 - PathExpression",
    ],
    "4,11:4,28 - SubExpression": Array [
      "4,13:4,17 - PathExpression",
    ],
    "6,4:6,11 - MustacheStatement": Array [
      "6,6:6,9 - PathExpression",
    ],
  },
  "scopes": Object {
    "2,4:8,9 - BlockStatement": Array [
      "tag",
    ],
    "3,2:7,11 - BlockStatement": Array [],
    "6,4:6,11 - MustacheStatement": Array [],
  },
}
`;

exports[` 2`] = `
Object {
  "blockPaths": Array [
    "1,3:1,7 - PathExpression",
  ],
  "componentsForImport": Array [],
  "parents": Object {
    "1,0:3,11 - BlockStatement": Array [
      "2,4:2,11 - MustacheStatement",
      "1,3:1,7 - PathExpression",
      "1,8:1,23 - PathExpression",
    ],
    "2,4:2,11 - MustacheStatement": Array [
      "2,6:2,9 - PathExpression",
    ],
  },
  "scopes": Object {
    "1,0:3,11 - BlockStatement": Array [
      "tag",
    ],
    "2,4:2,11 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta <MyComponent as |ctx|> {{ctx.result}} </MyComponent> 1`] = `
Object {
  "blockPaths": Array [
    "1,0:1,52 - PathExpression",
  ],
  "componentsForImport": Array [
    "my-component",
  ],
  "parents": Object {
    "1,0:1,52 - BlockStatement": Array [
      "1,23:1,37 - MustacheStatement",
      "1,0:1,52 - PathExpression",
    ],
    "1,23:1,37 - MustacheStatement": Array [
      "1,25:1,35 - PathExpression",
    ],
  },
  "scopes": Object {
    "1,0:1,52 - BlockStatement": Array [
      "ctx",
    ],
    "1,23:1,37 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta <MyComponent as |ctx|>{{#let ctx.result as |bar|}}{{bar}}{{/let}}</MyComponent> 1`] = `
Object {
  "blockPaths": Array [
    "1,0:1,79 - PathExpression",
    "1,25:1,28 - PathExpression",
  ],
  "componentsForImport": Array [
    "my-component",
  ],
  "parents": Object {
    "1,0:1,79 - BlockStatement": Array [
      "1,22:1,65 - BlockStatement",
      "1,50:1,57 - MustacheStatement",
      "1,0:1,79 - PathExpression",
    ],
    "1,22:1,65 - BlockStatement": Array [
      "1,50:1,57 - MustacheStatement",
      "1,25:1,28 - PathExpression",
      "1,29:1,39 - PathExpression",
    ],
    "1,50:1,57 - MustacheStatement": Array [
      "1,52:1,55 - PathExpression",
    ],
  },
  "scopes": Object {
    "1,0:1,79 - BlockStatement": Array [
      "ctx",
    ],
    "1,22:1,65 - BlockStatement": Array [
      "bar",
    ],
    "1,50:1,57 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta <input {{foo-bar this.baz a="1"}}> 1`] = `
Object {
  "blockPaths": Array [],
  "componentsForImport": Array [],
  "parents": Object {
    "1,7:1,33 - ElementModifierStatement": Array [
      "1,9:1,16 - PathExpression",
      "1,17:1,25 - PathExpression",
      "1,28:1,31 - StringLiteral",
    ],
  },
  "scopes": Object {
    "1,7:1,33 - ElementModifierStatement": Array [],
  },
}
`;

exports[`getClassMeta {{"1"}} 1`] = `
Object {
  "blockPaths": Array [],
  "componentsForImport": Array [],
  "parents": Object {
    "1,0:1,7 - MustacheStatement": Array [
      "1,2:1,5 - StringLiteral",
    ],
  },
  "scopes": Object {
    "1,0:1,7 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta {{#let foo as |bar|}}{{bar}}{{/let}} 1`] = `
Object {
  "blockPaths": Array [
    "1,3:1,6 - PathExpression",
  ],
  "componentsForImport": Array [],
  "parents": Object {
    "1,0:1,36 - BlockStatement": Array [
      "1,21:1,28 - MustacheStatement",
      "1,3:1,6 - PathExpression",
      "1,7:1,10 - PathExpression",
    ],
    "1,21:1,28 - MustacheStatement": Array [
      "1,23:1,26 - PathExpression",
    ],
  },
  "scopes": Object {
    "1,0:1,36 - BlockStatement": Array [
      "bar",
    ],
    "1,21:1,28 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta {{1}} 1`] = `
Object {
  "blockPaths": Array [],
  "componentsForImport": Array [],
  "parents": Object {
    "1,0:1,5 - MustacheStatement": Array [
      "1,2:1,3 - NumberLiteral",
    ],
  },
  "scopes": Object {
    "1,0:1,5 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta {{foo this.bar baz=this.boo}} 1`] = `
Object {
  "blockPaths": Array [],
  "componentsForImport": Array [],
  "parents": Object {
    "1,0:1,29 - MustacheStatement": Array [
      "1,2:1,5 - PathExpression",
      "1,6:1,14 - PathExpression",
      "1,19:1,27 - PathExpression",
    ],
  },
  "scopes": Object {
    "1,0:1,29 - MustacheStatement": Array [],
  },
}
`;

exports[`getClassMeta {{foo}} 1`] = `
Object {
  "blockPaths": Array [],
  "componentsForImport": Array [],
  "parents": Object {
    "1,0:1,7 - MustacheStatement": Array [
      "1,2:1,5 - PathExpression",
    ],
  },
  "scopes": Object {
    "1,0:1,7 - MustacheStatement": Array [],
  },
}
`;
